shim/test/module/unsafeRec.js

```diff
@@ -0,0 +1,50 @@
+import test from 'tape';
+import vm from 'vm';
+import { getSharedGlobalDescs } from '../../src/stdlib';
+import { createNewUnsafeRec, createNewUnsafeGlobalForBrowser } from '../../src/unsafeRec';
+
+test('createNewUnsafeRec', t => {
+  t.plan(5);
+
+  const unsafeRec = createNewUnsafeRec();
+  const { unsafeGlobal, sharedGlobalDescs, unsafeEval, unsafeFunction, allShims } = unsafeRec;
+  const expectedSharedGlobalDescs = getSharedGlobalDescs(unsafeGlobal);
+
+  t.ok(Object.isFrozen(unsafeRec));
+
+  t.deepEqual(Object.keys(sharedGlobalDescs), Object.keys(expectedSharedGlobalDescs));
+  t.equal(unsafeEval, unsafeGlobal.eval);
+  t.equal(unsafeFunction, unsafeGlobal.Function);
+  t.deepEqual(allShims, []);
+});
+
+test('createNewUnsafeGlobalForBrowser', t => {
+  t.plan(3);
+
+  const window = vm.runInNewContext('this');
```

Comment:
Let's see if I have this right:

The code in this test case assumes (without checking) that it is run on node. It then provides a minimal emulation of a browser environment in order to test browser-specific code when run on node.

What happens when this test is run on a browser? Should we package this emulation up in a testing utils library of some sort? Should we also have a minimal node emulation that runs in the browser, so that we can test node specific code when running tests on the browser?

In any case, it is worth a comment to explain your intentions and assumptions here.

Reply:
Currentlty, these tests only execute on node. Typically, npm libraries have unit tests running on node or on node + jsdom. We can't use jsdom because it doesn't support iframe.contentWindow().

In my mind, it would be beneficial to have a complete test coverage whether we develop on node or on the browser, withouth requiring to execute all tests on both platforms to detect a breakage.

I have not added browser testing yet, however I have added a guard in preparation.